mlflow/pyfunc/__init__.py

```diff
@@ -603,6 +604,12 @@ def _warn_dependency_requirement_mismatches(model_path):
             req_line = req.req_str
             mismatch_info = _check_requirement_satisfied(req_line)
             if mismatch_info is not None:
+                # Suppress databricks-feature-lookup warning for feature store cases
+                if (
+                    mismatch_info.package_name == DATABRICKS_FEATURE_LOOKUP
+                    and module_name == _DATABRICKS_FS_LOADER_MODULE
```

Comment:
Why do we need to match on model name?

Reply:
The warning should only be supressed for feature store models

Reply:
Is that covered by the package name check already, or are there cases where someone includes DB feature lookup as a dependency in a model that doesn't use DB feature store as a loader module?